cli/lock.ml

```diff
@@ -515,7 +515,7 @@ let extract_source_config ~adjustment ~opam_monorepo_cwd ~opam_files
 
 let raw_cli_args () =
   match Array.to_list Sys.argv with
-  | _bin :: "lock" :: args -> args
+  | _bin :: prefix :: args when String.starts_with ~prefix "lock" -> args
```

Comment:
I think this would lead to a weird case where `opam-monorepo locksmith` goes into this case.

I am somewhat confused how you got into this place, since I would've expected `opam-monorepo locksmith` not resolving to this module to start with.

Reply:
It will only match prefixes of "lock", not strings which "lock" is a prefix of. `cmdliner` accepts prefixes of subcommands already. This code here is a sanity check that the command was a `opam monorepo lock` subcommand when saving the command to the lockfile.

Reply:
Ah, you're right, I've misread the code. Honestly, I think the previous code is fine and we/Cmdliner should not accept prefix matches. If the user uses `opam monorepo l`  and we introduce a new command that starts with `l` then suddenly things that worked don't anymore because it is impossible to disambiguate. Which can be annoying in e.g. scripts like the CI.